CIPs/0001.md

```diff
@@ -0,0 +1,48 @@
+# CIP 1: Validator-run attestation service
+
+- Date: 2019-08-19
+- Author: @nambrot
+
+### Overview
+
+The [lightweight identity protocol](https://docs.celo.org/celo-codebase/protocol/identity) allows users to map (the hash of) their phone numbers to account addresses, greatly improving the user experience of sending value on the Celo platform.
+
+To do so in a decentralized manner, we rely on validators running what we currently call a verification pool to observe attestation requests and fullfill them by sending them via a text message provider like Twilio. However, the service we created expects celo-blockchain to do the decryption and attestation message signing which may not be ideal. Further, the verification pool should be agnostic to the infrastructure it is run on, whereas now it is tightly coupled to Google Cloud Platform products Firebase Cloud Function and Firebase Database.
+
+This CIP outlines work to get the identity protocol to a more realistic production setting to possibly test on a Games of stakes betanet.
+
+### Goals
+
+- Get closer to a realistic production setting
+- Make the validator experience of providing attestations as smooth as possible
+- Make the verification process reliable and performant
+
+### Proposed Solution
+
+We propose the creation of a service called the attestation service. Its features would be as follows:
+
+**Accept Reveals via HTTPS**
+We currently have a reveal transaction whose sole purpose it is to communicate an encrypted phone number (identifier) to the validator. That is unncessary to do on-chain. While we will no longer be able to leverage the DDoS prevention built in to the blockchain layer, we think the trade-off is worth it, as it will remove one of three transactions in the whole process, thus speed it up, and reduce gas costs.
```

Comment:
We have to somehow communicate the service endpoint URL, imo we should reuse the metadata endpoint from #4, what do you think @timmoreton